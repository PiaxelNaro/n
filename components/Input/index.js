// @flow
import React from 'react';
import styled from 'styled-components';
import FlexGrid from '../FlexGrid';

const RawInput = styled.input`
    width: 100%;
    border: none;
    background-color: transparent;
    outline: none;
    height: 22px;
`;

const FakeInput = styled.div`
    display: block;
    border: 1px solid #999;
    border-radius: 16px;
    box-shadow: inset 0 1px 2px 0 rgba(0, 0, 0, 0.2);
    background-color: #fff;
    padding: 2px 10px;
`;

const Input = ({
    value,
    onChange,
    onClick,
    placeholder,
    icon,
    width,
}: {
    value: any,
    onChange: string => any,
    onClick?: SyntheticEvent => any,
    placeholder?: string,
    icon: any,
    width: string,
}) => (
    <FakeInput style={{ width }}>
        <FlexGrid>
            <FlexGrid.Item>
                <RawInput
                    value={value}
                    onChange={e => onChange(e.target.value)}
                    onClick={onClick}
                    placeholder={placeholder}
                />
            </FlexGrid.Item>
            {icon}
        </FlexGrid>
    </FakeInput>
);

Input.defaultProps = {
    onClick: null,
    placeholder: '',
};

export default Input;
