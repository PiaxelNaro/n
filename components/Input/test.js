// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Input from '.';

describe('<Input>', () => {
    it('renders input field and handle change events', () => {
        const spy = jest.fn();
        const input = shallow(
            <Input
                value="bar"
                onChange={spy}
                onClick={() => null}
                placeholder="Hey"
                icon="l"
                width="200px"
            />
        );
        expect(input).toMatchSnapshot();

        input.find('Input__RawInput').simulate('change', { target: { value: 'bar' } });
        expect(spy).toHaveBeenCalledWith('bar');
    });
});
