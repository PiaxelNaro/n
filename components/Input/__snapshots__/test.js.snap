// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> renders input field and handle change events 1`] = `
<Input__FakeInput
  style={
    Object {
      "width": "200px",
    }
  }
>
  <FlexGrid>
    <FlexGrid.Item
      flex={1}
    >
      <Input__RawInput
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Hey"
        value="bar"
      />
    </FlexGrid.Item>
    l
  </FlexGrid>
</Input__FakeInput>
`;
